folio-spring-i18n/src/main/java/org/folio/spring/i18n/model/TranslationFile.java

```diff
@@ -0,0 +1,148 @@
+import lombok.With;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.core.io.Resource;
+
+/**
+ * A class to represent a file that contains translation information.  This file will have a name
+ * identifying its associated locale with its contents being an ICU dictionary.
+ *
+ * <p>This class also contains several utility methods for parsing filenames and aggregating
+ * {@code TranslationFile} objects.</p>
+ */
+@Value
+@Log4j2
+@AllArgsConstructor
+public class TranslationFile {
+
+  /**
+   * The String denoting that a language or country is unknown from {@link getParts}.
+   */
+  public static final String UNKNOWN_PART = "*";
+
+  private static final int MAX_FILENAME_PARTS = 2;
+  private static final String JSON_FILE_SUFFIX = ".json";
+
+  /**
+   * The resources backing the translation files.
+   */
+  private List<Resource> resources;
+
+  /**
+   * Get the map of patterns from this file.
+   *
+   * @return the map of patterns -&gt; ICU format strings
+   */
+  public Map<String, String> getPatterns() {
+    Map<String, String> result = new HashMap<>();
+    ObjectMapper objectMapper = new ObjectMapper();
+
+    for (Resource resource : resources) {
+      try {
+        Map<String, String> moduleMap = objectMapper.readValue(
+          resource.getInputStream(),
+          new TypeReference<Map<String, String>>() {}
+        );
+
+        String moduleName = Path
+          .of(resource.getURL().getPath())
+          .getParent()
+          .getFileName()
+          .toString();
+
+        moduleMap.forEach((key, value) ->
+          result.put(moduleName + "." + key, value)
+        );
+      } catch (IOException e) {
+        log.error(
+          "Could not open/read translation file {}; will use fallback instead",
+          resource.getDescription(),
+          e
+        );
+      }
+    }
+
+    return result;
+  }
+
+  /**
+   * Get the parts of this filename as a {@link LanguageRegionPair LanguageRegionPair}.
+   *
+   * <p>If one (or either) is unknown, {@code UNKNOWN_PART} will be returned.</p>
+   *
+   * @return a {@link LanguageRegionPair LanguageRegionPair}
+   */
+  public LanguageRegionPair getParts() {
+    // Spring's getFilename returns only the last part, so no need to split the path
+    return getParts(resources.get(0).getFilename());
+  }
+
+  /**
+   * Get the parts of a filename as an array of two elements: {@link LanguageRegionPair LanguageRegionPair}.
+   * These will be converted to lowercase, if applicable.
+   *
+   * <p>If one (or either) is unknown, {@code UNKNOWN_PART} will be returned.</p>
+   *
+   * @param filename the filename to parse
+   * @return a {@link LanguageRegionPair LanguageRegionPair}
+   */
+  public static LanguageRegionPair getParts(String filename) {
+    LanguageRegionPair result = new LanguageRegionPair(
+      UNKNOWN_PART,
+      UNKNOWN_PART
+    );
+
+    if (filename == null) {
+      return result;
+    }
+
+    if (filename.endsWith(JSON_FILE_SUFFIX)) {
```

Comment:
What about other file formats? Is there a case where we should try to process not-json format?

Reply:
UI modules/Lokalise currently only support the JSON format

Reply:
So if we have not json file then some exception should be thrown.